cereus/src/main/java/com/cactus/entities/User.java

```diff
@@ -20,14 +26,21 @@
      * @param name A String containing the User's name.
      * @param username A String containing the User's username.
      * @param password A String containing the User's password.
+     * @param token A String containing a token to signify User was successfully retrieved from server
+     * @param id A long value serving as an identifier for the User.
      */
-    public User(String name, String username, String password){
+
+    public User(String name, String username, String password, String token, long id){
         this.name = name;
         this.username = username;
         this.password = password;
-        this.id = 0L;
+        this.token = token;
+        this.id = id;
     }
 
+    public User(){
+        super();
```

Comment:
Since `Entity` is an interface currently, calling `super()` has no effect, all it does is call the `Object` constructor I think.

Reply:
and if we make entity an abstract class, it will try to call the constructor of entity

Reply:
The empty User constructor is important for the Jackson library to create the User object from the HttpResponse string. This, along with all the getters and setters for the attributes. It's also why I had to remove the final attribute of the username, cause otherwise the Jackson library won't work.

Reply:
Right, so if you need the empty constructor it's fine to just leave the body empty, you don't need the super constructor call.

Reply:
Yeah I thought it needed the super call, but the code is working without it so I will take out the super call.